src/fidesops/schemas/saas/saas_config.py

```diff
@@ -0,0 +1,142 @@
+    data_path: Optional[str]  # defaults to collection name if not specified
+    preprocessor: Optional[Strategy]
+    postprocessor: Optional[List[Strategy]]
+    pagination: Optional[Strategy]
+
+
+class Endpoint(BaseModel):
+    """An collection of read/update/delete requests which corresponds to a FidesopsDataset collection (by name)"""
+
+    name: str
+    requests: Dict[Literal["read", "update", "delete"], Request]
+
+
+class ConnectorParam(BaseModel):
+    """Used to define the required parameters for the connector (user-provided and constants)"""
+
+    name: str
+    default_value: Optional[Any]
+    from_user: bool
+    is_secret: bool
+
+
+class ConnectorParamRef(BaseModel):
+    """A reference to a value in the connector params (by name)"""
+
+    connector_param: str
+
+
+class ClientConfig(BaseModel):
+    """Definition for an authenticated base HTTP client"""
+
+    protocol: str
+    host: Union[
+        str, ConnectorParamRef
+    ]  # can be defined inline or be a connector_param reference
+    authentication: Strategy
+
+
+class SaaSConfig(BaseModel):
+    """
+    Used to store endpoint and param configurations for a SaaS connector.
+    This is done to separate the details of how to make the API calls
+    from the data provided by a given API collection.
+
+    The required fields for the config are converted into a Dataset which is
+    merged with the standard Fidesops Dataset to provide a complete set of dependencies
+    for the graph traversal
+    """
+
+    fides_key: FidesOpsKey
+    name: str
+    description: str
+    version: str
+    connector_params: List[ConnectorParam]
+    client_config: ClientConfig
+    endpoints: List[Endpoint]
+    test_request: Request
+
+    @property
+    def top_level_endpoint_dict(self) -> Dict[str, Endpoint]:
+        """Returns a map of endpoint names mapped to Endpoints"""
+        return {endpoint.name: endpoint for endpoint in self.endpoints}
+
+    def get_graph(self) -> Dataset:
+        """Converts endpoints to a Dataset with collections and field references"""
+        collections = []
+        for endpoint in self.endpoints:
+            fields = []
+            # TODO parametrize this method to account for update and delete
+            for param in endpoint.requests["read"].request_params or []:
+                if param.references:
+                    references = []
+                    for reference in param.references:
+                        first, *rest = reference.field.split(".")
+                        references.append(
+                            (
+                                FieldAddress(reference.dataset, first, *rest),
+                                reference.direction,
+                            )
+                        )
+                    fields.append(ScalarField(name=param.name, references=references))
+                if param.identity:
+                    fields.append(ScalarField(name=param.name, identity=param.identity))
```

Comment:
I had an earlier question about whether something could have an identity and a reference - if this shouldn't happen, I think we should add validation for it to prevent that from happening.

Reply:
Will include this validation in the API endpoint ticket.